src/PKSim.Core/Model/RandomPopulationFactory.cs

```diff
@@ -105,21 +121,28 @@ public Task<RandomPopulation> CreateFor(RandomPopulationSettings populationSetti
                      throw new CannotCreatePopulationWithConstraintsException(_reportGenerator.StringReportFor(populationSettings));
 
                   //create a new individual based on population settings defined by the user
-                  updateCurrentIndividualFromSettings(populationSettings, currentIndividual, allDistributedParameters, allBaseDistributedParameters, currentGender, randomPopulation.RandomGenerator);
+                  updateCurrentIndividualFromSettings(populationSettings, baseIndividual, currentIndividual, allDistributedParameters, allBaseDistributedParameters, currentGender, randomPopulation.RandomGenerator);
 
-                  bool success = tryRandomize(currentIndividual, populationSettings, allCurrentIndividualParameters, randomPopulation.RandomGenerator);
-                  if (!success) continue;
+                  var success = tryRandomize(currentIndividual, populationSettings, allCurrentIndividualParameters, randomPopulation.RandomGenerator);
+                  if (!success)
+                     continue;
+
+                  success = diseaseStateImplementation.ApplyForPopulationTo(currentIndividual);
```

Comment:
Second Hook. Instead of Apply that overwrites also the formula, I need to have a method that only overwrites the value in the indivudal so that I can reset them later